[WFGP-230] Using resolveAll to resolve jboss modules

spyrkob opened the pull request:
Issue: https://issues.redhat.com/browse/WFGP-230

If 'bulk-resolve-artifacts' option is specified, module templates are scanned before being processing to build a list of required artifacts.
All the required artifacts are then resolved in a single Maven operation rather then individual requests.

Note: depends on Galleon API change https://github.com/wildfly/galleon/pull/305
Note 2: related channels change: https://github.com/wildfly-extras/wildfly-channel/pull/75

jfdenise commented on galleon-plugins/src/main/java/org/wildfly/galleon/plugin/WfInstallPlugin.java:
```diff
@@ -543,6 +564,64 @@ public void resolve(MavenArtifact artifact) throws ProvisioningException {
+            b.setVersionRange(a.getVersionRange());
+            artifactCache.put(b, a);
+        }
+        maven.resolveAll(allModuleArtifacts);
```
If we have a pre-populated cache, should be: maven.resolveAll(artifactCache.keySet());
This implies that the maven.resolveAll method signature be changed from List<MavenArtifact> to Collection<MavenArtifact>.

spyrkob replied:
that would put the resolved artifacts as the keys. I think that would prevent them from matching when they're resolved in ModuleProcessors.

spyrkob replied:
nevermind, didn't click that it's related to earlier comment. I'll take a look

spyrkob commented:
I think that should address the changes.
I added a cache for templates and changed the way artifact cache is populated. Also added a small refactor to processing boolean flags.

Updated  https://github.com/wildfly/galleon/pull/305 with new resolveAll method as well.

jfdenise commented:
@spyrkob , galleon 5.0.2.Final has been released, you can update the PR. Thanks.

spyrkob commented:
Updated Galleon version and squashed commits

jfdenise commented:
@spyrkob , thank-you, merged!
